packages/react/src/components/Authenticator/hooks/useAuthenticator/index.tsx

```diff
@@ -179,12 +150,21 @@ export const useAuthenticator = (selector?: Selector) => {
     return areArrayValuesEqual(prevDepsArray, nextDepsArray);
   };
 
-  const state = useSelector(service, xstateSelector, comparator);
+  const facade = useSelector(
+    service,
+    xstateSelector,
+    comparator,
+    /**
+     * Below parameter was needed to avoid "The result of getSnapshot should be
+     * cached to avoid an infinite loop" error on React 18
+     */
+    (interpreter) => interpreter.getSnapshot()
+  );
 
   return {
-    ...getFacade(state),
+    ...facade,
     /** @deprecated For internal use only */
-    _state: state,
+    _state: service.getSnapshot(),
```

Comment:
Assuming `service.getSnapshot` returns a new object on each call, does computing it for each call to `useAuthenticator` have any impact on re-renders? Follow up question, from what i understand, since our usage of `_state` is not for handling dynamic values, maybe we should consider assigning its value to a ref that only computes on the first call to `useAuthenticator`?

Reply:
Great catch! Here was where I was not feeling 100% too.

(1) This would not trigger more re-renders, but yes this is additional computation that we didn't have. This was now needed that we're using `selector` to get `facade` directly. This does not change the runtime complexity though, because `getSnapshot` was already getting called through `useSelector`.

 A long term solution would be to not depend on `_state` at all, or (2):

(2) I like the idea! I do want to scope it out of this PR though, because I want to audit and make sure what you said about `_state` is true.  e.g.

https://github.com/aws-amplify/amplify-ui/blob/3729ac84f3dd6b81e28e055edab41f9761384f9a/packages/react/src/components/Authenticator/FederatedSignIn/FederatedSignIn.tsx#L11

depends on dynamic value. It might be worth more effort to investigate expanding `useAuthenticator` facade values so that `_state` doesn't get used like that there.

Reply:
Took a quick look through the implementation, `getSnapshot` just returns `_state` underneath the interpreter and does not return a new object. 

https://github.com/statelyai/xstate/blob/c0b787d2a2334665af0ecb4ebc1ce5653fa3725c/packages/core/src/interpreter.ts#L1394

So this is not an expensive operation. This explains why we were having `ref` comparison problems too!